src/timeaxis/timegrid.jl

```diff
@@ -287,6 +299,45 @@ Base.resize!(tg::TimeGrid{T,P,:infinite}, n::Int) where {T,P} = tg
 
 Base.diff(tg::TimeGrid{T,P,:finite}) where {T,P} = fill(tg.p, tg.n - 1)
 
+###############################################################################
+#  Intersect
+###############################################################################
+function Base.intersect(tg1::TimeGrid, tgs...)
+    for tg in tgs
+        tg1 = intersect(tg1, tg)
+    end
+    return tg1
+end
+# TODO 
+# Consider checking if they are equal
+function intersect(tg_inf::TimeGrid{T,P,:infinite}, 
+                   tg_fin::TimeGrid{T,P,:finite})::TimeGrid{T,P,:finite} where {T,P}
+    @assert tg_fin.p % tg_inf.p
+    if tg_fin[end] < tg_inf.o
+        return TimeGrid(tg_fin.o, tg_fin.p, 0)
+    end
+    new_o = max(tg_fin.o, tg_inf.o)
```

Comment:
Is it okay with you, if I rename `new_o` to `o′` someday?
Personally, I prefer shorter name. :p

Reply:
no problem